### PR TITLE
glib: Variant wrapper types for handles, object paths and signatures

### PR DESCRIPTION
**Pull request by sdroege**
CC @SeaDve

### CODE REVIEW

**Comment by SeaDve** [comment]
Thanks for looking at the issue! LGTM

**Comment by sdroege** [comment]
Any other opinions here @bilelmoussaoui @GuillaumeGomez ?

File: glib/src/variant.rs [hunk 1]
```diff
@@ -1757,6 +1757,188 @@ impl<A: AsRef<[T]>, T: FixedSizeVariantType> ToVariant for FixedSizeVariantArray
     }
 }
 
+/// A wrapper type around `Variant` handles.
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct Handle(pub i32);
```
Comment (bilelmoussaoui) [inline glib/src/variant.rs:1762]: Any reason why the inner value is public?

Comment (sdroege) [reply glib/src/variant.rs:1762]: Just because "why not" :) There's no problem giving direct access to it

Comment (bilelmoussaoui) [reply glib/src/variant.rs:1762]: I guess I have no idea in which context you would use this. Making it public allows people to construct a new `Handle` that might contain a garbage value but then we also have a From implementation so :)

Comment (sdroege) [reply glib/src/variant.rs:1762]: It's a plain integer that is an index into an out of band transmitted array (of which you know the size). No problem here

Comment (sdroege) [reply glib/src/variant.rs:1762]: Also you can easily parse any kind of integer as a handle from a string anyway.

**Review by bilelmoussaoui — APPROVED** [review]

**Pull request merged**